packages/cli-tool/src/add-ons/ui-framework/bootstrap.ts

```diff
@@ -0,0 +1,76 @@
+import * as fs from 'node:fs';
+
+import { cli } from 'cli-ux';
+
+import runCommand from '../../helpers/child-process';
+import {
+  addLinesToFileAfterMatchedLine,
+  lineFinderFuncType,
+} from '../../helpers/file-editor';
+
+export const addBootstrapFileStructure = (
+  appName: string,
+): Promise<boolean> => {
+  return new Promise((resolve, reject) => {
+    cli.info('Starting: ', `./${appName}/.add-ons/bootstrap`);
+
+    try {
+      const destPath = `./${appName}/src/assets/stylesheets/vendor/bootstrap/`;
+      fs.renameSync(`./${appName}/.add-ons/bootstrap/`, destPath);
+
+      resolve(true);
+    } catch (error) {
+      cli.info(
+        'Error while copying vendor files for Bootstrap:',
+        error as string,
+      );
+
+      reject(error);
+    }
+  });
+};
+
+const bootstrapImportLocationFinder: lineFinderFuncType = (
+  lines: string[],
+): number => {
+  return lines.indexOf('// Dependencies');
+};
+
+// In index.scss, Replace "// Dependencies" by `// Dependencies \n@import 'vendor/bootstrap';` in application.scss
+const addBootstrapScssUseLine = (appName: string): Promise<boolean> => {
+  return new Promise((resolve, reject) => {
+    cli.info('Insert bootstrap scss import.');
+
+    try {
+      const indexScssPath = `./${appName}/src/assets/stylesheets/application.scss`;
+      const lineToAdd = `@use 'src/assets/stylesheets/vendor/bootstrap';`;
+
+      addLinesToFileAfterMatchedLine(
+        indexScssPath,
+        bootstrapImportLocationFinder,
+        [lineToAdd],
+      );
+
+      resolve(true);
+    } catch (error) {
+      cli.info(
+        'Error while adding the bootstrap lib import in index.scss:',
+        error as string,
+      );
+
+      reject(error);
+    }
+  });
+};
+
+const installNpmPackage = (appName: string): Promise<boolean> => {
+  return runCommand('npm', ['install', 'bootstrap@^5.1.3'], `./${appName}/`);
```

Comment:
Should we move 'bootstrap@^5.1.3' to a constant? just like you did on https://github.com/nimblehq/react-templates/pull/107/files#diff-d8c30ea5a4cdaf36504485b1b439a19c87667f1c5d1951ea7356ae7a48e3ffb2R8
Then later we can easily update the version by looking for that constant and changing version number.
Or we can create a file to store all NPM packages' versions, so we will refer to that file in each add-on 🤔

Reply:
✅ c09bf28